server/apps/publish/formatters/newsml_1_2_formatter.py

```diff
@@ -0,0 +1,163 @@
+        news_management = SubElement(news_item, "NewsManagement")
+        SubElement(news_management, 'NewsItemType', {'FormalName': 'News'})
+        SubElement(news_management, 'FirstCreated').text = \
+            article['firstcreated'].strftime('%Y%m%dT%H%M%S+0000')
+        SubElement(news_management, 'ThisRevisionCreated').text = \
+            article['versioncreated'].strftime('%Y%m%dT%H%M%S+0000')
+        SubElement(news_management, 'Status', {'FormalName': article['pubstatus']})
+        SubElement(news_management, 'Urgency', {'FormalName': str(article['urgency'])})
+        if article['state'] == 'corrected':
+            SubElement(news_management, 'Instruction', {'FormalName': 'Correction'})
+        else:
+            SubElement(news_management, 'Instruction', {'FormalName': 'Update'})
+
+    def __format_news_component(self, article, news_item):
+        news_component = SubElement(news_item, "NewsComponent")
+        main_news_component = SubElement(news_component, "NewsComponent")
+        SubElement(main_news_component, 'Role', {'FormalName': 'Main'})
+        self.__format_news_lines(article, main_news_component)
+        self.__format_rights_metadata(article, main_news_component)
+        self.__format_descriptive_metadata(article, main_news_component)
+        self.__format_abstract(article, main_news_component)
+        self.__format_body(article, main_news_component)
+
+    def __format_news_lines(self, article, main_news_component):
+        news_lines = SubElement(main_news_component, "NewsLines")
+        SubElement(news_lines, 'Headline').text = article.get('headline', '')
+        SubElement(news_lines, 'ByLine').text = article.get('byline', '')
+        SubElement(news_lines, 'DateLine').text = article.get('dateline', '')
+        SubElement(news_lines, 'CreditLine').text = article.get('creditline', '')
+        SubElement(news_lines, 'KeywordLine').text = article.get('slugline', '')
+
+    def __format_rights_metadata(self, article, main_news_component):
+        rights_metadata = SubElement(main_news_component, "RightsMetadata")
+        copyright = SubElement(rights_metadata, "Copyright")
+        SubElement(copyright, 'CopyrightHolder').text = article.get('source', article.get('original_source', ''))
+        SubElement(copyright, 'CopyrightDate').text = self.now.strftime("%Y")
+
+        usage_rights = SubElement(rights_metadata, "UsageRights")
+        SubElement(usage_rights, 'UsageType').text = self.USAGE_TYPE
+        SubElement(usage_rights, 'Geography').text = article.get('place', article.get('located', ''))
+        SubElement(usage_rights, 'RightsHolder').text = article.get('source', article.get('original_source', ''))
+        SubElement(usage_rights, 'Limitations').text = self.LIMITATIONS
+        SubElement(usage_rights, 'StartDate').text = self.now.strftime('%Y%m%dT%H%M%S+0000')
+        SubElement(usage_rights, 'EndDate').text = self.now.strftime('%Y%m%dT%H%M%S+0000')
```

Comment:
This datetime format specification occurs at multiple places, perhaps move it to some class level "constant"?